src/operator/random/pdf_op.cc

```diff
@@ -0,0 +1,292 @@
+      [[0.36787945 0.13533528 0.04978707]]
+
+)code");
+}
+
+inline std::string poisson_desc() {
+  return std::string(R"code(Computes the value of the PDF of *sample* of
+Poisson distributions with parameters *lam* (rate).
+
+Given *lam* of dimension *n* and *sample* of dimension *n+1*, produces
+output of dimension *n+1* such that each *n*-dimensional index *i* in
+the output array holds the PDFs of the samples at index *i* in *sample*,
+parameterized by the value of *lam* at index *i*.
+
+Examples::
+
+    random_pdf_poisson(sample=[[0,1,2,3]], lam=[1]) =
+        [[0.36787945 0.36787945 0.18393973 0.06131324]]
+
+    sample = [[0,1,2,3],
+              [0,1,2,3],
+              [0,1,2,3]]
+
+    random_pdf_poisson(sample=sample, lam=[1,2,3]) =
+        [[0.36787945 0.36787945 0.18393973 0.06131324]
+         [0.13533528 0.27067056 0.27067056 0.18044704]
+         [0.04978707 0.14936121 0.22404182 0.22404182]]
+)code");
+}
+
+inline std::string negative_binomial_desc() {
+  return std::string(R"code(Computes the value of the PDF of samples of
+negative binomial distributions with parameters *k* (failure limit) and *p* (failure probability).
+
+Given *k* and *p* of dimension *n* and *sample* of dimension *n+1*, produces
+output of dimension *n+1* such that each *n*-dimensional index *i* in
+the output array holds the PDFs of the samples at index *i* in *sample*,
+parameterized by the values of *k* and *p* at index *i*.
+
+Examples::
+
+    random_pdf_negative_binomial(sample=[[1,2,3,4]], k=[1], p=a[0.5]) =
+        [[0.25    0.125   0.0625  0.03125]]
+
+    # Note that k may be real-valued
+    sample = [[1,2,3,4],
+              [1,2,3,4]]
+    random_pdf_negative_binomial(sample=sample, k=[1, 1.5], p=[0.5, 0.5]) =
+        [[0.25       0.125      0.0625     0.03125   ]
+         [0.26516506 0.16572815 0.09667476 0.05437956]]
+)code");
+}
+
+inline std::string generalized_negative_binomial_desc() {
+  return std::string(R"code(Computes the value of the PDF of *sample* of
+generalized negative binomial distributions with parameters *mu* (mean)
+and *alpha* (dispersion).  This can be understood as a reparameterization of
+the negative binomial, where *k* = *1 / alpha* and *p* = *1 / (mu \* alpha + 1)*.
+
+Given *k* and *p* of dimension *n* and *sample* of dimension *n+1*, produces
+output of dimension *n+1* such that each *n*-dimensional index *i* in
+the output array holds the PDFs of the samples at index *i* in *sample*,
+parameterized by the values of *k* and *p* at index *i*.
+
+Examples::
+
+    random_pdf_generalized_negative_binomial(sample=[[1, 2, 3, 4]], alpha=[1], mu=[1]) =
+        [[0.25    0.125   0.0625  0.03125]]
+
+    sample = [[1,2,3,4],
+              [1,2,3,4]]
+    random_pdf_generalized_negative_binomial(sample=sample, alpha=[1, 0.6666], mu=[1, 1.5]) =
+        [[0.25       0.125      0.0625     0.03125   ]
+         [0.26517063 0.16573331 0.09667706 0.05437994]]
+)code");
+}
+
+inline std::string dirichlet_desc() {
```

Comment:
I don't think we have a function to draw samples from a dirichlet distribution? this only computes the pdf of the distribution?

Reply:
Updated the commit message to clarify that, unlike the other distributions, we don't have a Dirichlet sampler yet.